module/os/operation_siren.py

```diff
@@ -81,11 +82,13 @@ def globe_goto(self, zone, types=('SAFE', 'DANGEROUS'), refresh=False, stop_if_s
         self.globe_focus_to(zone)
         if stop_if_safe:
             pinned = self.pinned_to_name(self.get_zone_pinned())
-            if pinned == 'SAFE':
+            if pinned == 'SAFE' or \
+              (self.zone_has_switch() and self.zone_type_select(types='SAFE', exclusive=True)):
```

Comment:
Add a new method `zone_has_safe` in globe_operation.py to replace this. Not to modify `zone_type_select`.

Reply:
So back out all changes in `zone_type_select`, including this part? 
```
        found = True # This found flag to determine whether entered the button is None clause
        for _ in range(3):
            self.ui_click(ZONE_SWITCH, appear_button=self.is_zone_pinned, check_button=self.is_in_zone_select,
                          skip_first_screenshot=True)

            selection = self.get_zone_select()
            logger.attr('Zone_selection', selection)
            button = get_button(selection)
            if button is None:
                logger.warning('No such zone type to select, fallback to default')
                types = ('SAFE', 'DANGEROUS')
                button = get_button(selection)
                found = False # Whether to return False since defaulted to a select
```
Since this flag is the important mechanism as I rely on the returning boolean whether actually found or just found something to take its place so can revert to pinned state.  Without it, based on testing always returned `True` since zone will likely always have the `DANGEROUS` option

Duplicate what I changed here and just place under `zone_has_safe` method signature?

Reply:
It's not neccessary to call `zone_type_select`, You can implement like:
```python
def zone_has_safe(self):
    if self.pinned_to_name(self.get_zone_pinned()) == 'SAFE':
        return True
    else:
        self.ui_click(ZONE_SWITCH, appear_button=self.is_zone_pinned, check_button=self.is_in_zone_select,
                              skip_first_screenshot=True)
        flag = SELECT_SAFE in self.get_zone_select():
        self.ui_click(button, check_button=self.is_zone_pinned, offset=self._zone_select_offset,
                          skip_first_screenshot=True)
        return flag
```